### PR TITLE
fix(fw-update): multiple fixes

### LINKED ISSUE #2541: Onboarding - Device Firmware update process logic
**Info:**
 - OS: Nixos + Windows 10
 - Wallet Version: 20.10.1 app, commit 426ce98fbcb16af057d8b2f5d636f88f4e374a23
 - Firmware Version: 2.3.1 model T, revision c6b2580cd
 
**Describe the bug**
Looks like Suite is not detecting FW version correctly during model T FW update When in bootloader mode.
Please see video. I was able to replicate it via Nixos and Windows app builds.
No idea if it shows booloader version /i have only 2.0.3 BL devices/

I tried to check same process with T1 model but it says. Error: Cannot read property 'code' of null
See screenshot, When I press retry it only shows Onboarding modal without any kind of info

**Steps to reproduce:**
1. Start Suite app and connect empty device with older FW
2. Start onboarding and FW update
3. When asked to install press X on device and continue with process

**Expected result:**
Suite should say firmware version according to installed FW

**Actual result:**
Suite says firmware version is 2.0.3 , and not 2.3.1

**Screenshots:**
Video: https://www.youtube.com/watch?v=fzOasmoNDDQ&feature=youtu.be

image

image

**T1**
image

**T1**
image

### LINKED ISSUE #2627: Make "Yes, I do!" clickable during firmware update
**Info:**
- OS: NixOS
- Browser: desktop app
- Suite Version: https://staging-suite.trezor.io/ - release/2020-10 - 9995926
- Firmware Version: 1.9.0
 
**Describe the bug**
I have to click the checkbox, not just the "Yes, I do!" which is a bit annoying :).

**Screenshots:**
a

### LINKED ISSUE #2650: Empty screen after update succeeds.
**Info:**
 - OS: macOS 10.14.6
 - Suite Version: macOS, 20.10.1
 - Firmware Version: Model T, update to 2.3.4 
 
**Describe the bug**
Empty screen in Suite after the firmware update finishes. Clears after the device is unlocked. Not a serious issue, it just looks like Suite is stuck (see screenshot).

**Steps to reproduce:**
1. Update firmware
2. Finish updating firmware, device restarts
3. Suite shows empty screen

**Expected result:**
Expected a screen with further instructions, like unlock your device to continue

**Actual result:**
Empty screen

**Screenshots:**
Screenshot 2020-10-15 at 21 00 43

**Note(s):**
Nice bug reporting template!

### PR DESCRIPTION
**Pull request by mroz22**
Changes:
- [x] change how retry button works, now it does not reset firmware update internal state but tries to install firmware directly. If there is error, in onboarding, back button resets internal firmware update state.

Fixes:
- [x] fix #2541 (firmware version vs bootloade version, UI problem)
- [x] fix #2650 by showing success screen without waiting for device to reconnect
- [x] fix #2627 make checkbox text clickable
- [x] [error code "null"](https://user-images.githubusercontent.com/31506317/95347757-8a79a080-08bd-11eb-99b4-306357c994e6.png). After cancellation of fw update on device. As this happens somewhere lower, this PR contains only a workaround and a separate issue is made for this. https://github.com/trezor/trezor-suite/issues/2659

Added: 
- e2e test for retry button behavior 

Not fixing at the moment:
- https://github.com/trezor/trezor-suite/issues/2456 requires broader discussion. luckily edge case

QA:
- [ ] Asana request (will file it after changes resulting from review)
- [ ] approval by QA

### CODE REVIEW

File: packages/suite/src/components/firmware/OnboardingInitial.tsx [hunk 1]
```diff
@@ -91,13 +101,11 @@ const BottomBar = () => {
     if (!device?.connected || !device?.features) return null;
 
     if (device?.firmware === 'none') {
-        return (
-            <Button onClick={() => firmwareUpdate()} data-test="@firmware/start-button">
-                <Translation id="TR_INSTALL" />
-            </Button>
-        );
+        return <InstallButton onClick={() => firmwareUpdate()} />;
```
Comment (szymonlesisz) [inline packages/suite/src/components/firmware/OnboardingInitial.tsx]: `onClick={firmwareUpdate}`

File: packages/suite/src/components/firmware/OnboardingInitial.tsx [hunk 2]
```diff
@@ -34,10 +41,13 @@ const Body = () => {
         );
     }
 
+    // after firmware none because such device reports as in bootloader mode
+    if (device?.mode === 'bootloader') return <ReconnectInNormalStep.Body />;
```
Comment (szymonlesisz) [inline packages/suite/src/components/firmware/OnboardingInitial.tsx]: rm ?

File: packages/suite/src/components/firmware/OnboardingInitial.tsx [hunk 3]
```diff
@@ -91,13 +101,11 @@ const BottomBar = () => {
     if (!device?.connected || !device?.features) return null;
```
Comment (szymonlesisz) [inline packages/suite/src/components/firmware/OnboardingInitial.tsx]: rm ?

File: packages/suite/src/components/firmware/OnboardingInitial.tsx [hunk 4]
```diff
@@ -91,13 +101,11 @@ const BottomBar = () => {
...
     if (device?.firmware === 'none') {
-        return (
-            <Button onClick={() => firmwareUpdate()} data-test="@firmware/start-button">
-                <Translation id="TR_INSTALL" />
-            </Button>
-        );
+        return <InstallButton onClick={() => firmwareUpdate()} />;
     }
 
+    if (device?.mode === 'bootloader') return null;
```
Comment (szymonlesisz) [inline packages/suite/src/components/firmware/OnboardingInitial.tsx]: rm `?`

File: packages/suite/src/reducers/firmware/firmwareReducer.ts [hunk 5]
```diff
@@ -52,10 +51,15 @@ const firmwareUpdate = (state: FirmwareUpdateState = initialState, action: Actio
         switch (action.type) {
             case FIRMWARE.SET_UPDATE_STATUS:
                 draft.status = action.payload;
+                if (action.payload === 'started') {
+                    draft.error = undefined;
```
Comment (szymonlesisz) [inline packages/suite/src/reducers/firmware/firmwareReducer.ts:55]: delete draft.error

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx [hunk 6]
```diff
@@ -24,13 +24,15 @@ const FirmwareStep = ({
     goToPreviousStep,
     goToNextStep,
     resetReducer,
+    firmwareUpdate,
 }: Props) => {
     const getComponent = () => {
+        console.log(firmware.error, firmware.status);
```
Comment (szymonlesisz) [inline packages/suite/src/views/onboarding/steps/Firmware/index.tsx]: rm console?

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx [hunk 7]
```diff
@@ -24,13 +24,15 @@ const FirmwareStep = ({
...
         // edge case 1 - error
         if (firmware.error) {
             return {
                 Body: <ErrorStep.Body />,
-                BottomBar: <RetryButton onClick={() => resetReducer()} />,
+                BottomBar: <RetryButton onClick={() => firmwareUpdate()} />,
```
Comment (szymonlesisz) [inline packages/suite/src/views/onboarding/steps/Firmware/index.tsx]: onClick={firmwareUpdate}

**Review by keraf — CHANGES_REQUESTED** [review]
No additional comments from me, just what Szymon said :slightly_smiling_face:

File: packages/suite/src/components/firmware/OnboardingInitial.tsx (hunk 3 above)
Comment (mroz22) [reply packages/suite/src/components/firmware/OnboardingInitial.tsx]: Its correct imho 
```
// unacquired, disconnected.
if (!device?.connected || !device?.features) return null;

// handle empty device, this shows bootloader mode true but this is expected
if (device?.firmware === 'none') {
   return <InstallButton onClick={firmwareUpdate} />;
}

// later handle outdated firmwares. device should not be in bootloader mode in the first step, so // if it is, don't render any "install" button
if (device?.mode === 'bootloader') return null;
```

Comment (mroz22) [reply packages/suite/src/components/firmware/OnboardingInitial.tsx]: aha, you probably meant optional chaining operator :D. if used on !device?.connected, it is not needed on !device?.features. that should be right.

File: packages/suite/src/reducers/firmware/firmwareReducer.ts (hunk 5 above)
Comment (mroz22) [reply packages/suite/src/reducers/firmware/firmwareReducer.ts:55]: Just checked that it really works 

image

**Comment by mroz22** [comment]
I am not sure what is better, regarding the model T pin case. 

1. Let UI hang on "waiting for your device to reboot" and update screen only after user enters PIN on device. Maybe we could check whether device has PIN and for such cases update the massage "waiting for your device to reboot, please enter PIN on your device"
1. Set success screen directly for model T devices. (current state of affairs in this PR)

**Comment by szymonlesisz** [comment]
> I am not sure what is better, regarding the model T pin case.

`1` looks like a feature since we have `2` already. you can create another PR for that let's merge this one first

**Review by szymonlesisz — APPROVED** [review]
